plugins/ui-update/components/UIDetails.tsx

```diff
@@ -1,116 +1,39 @@
 import * as React from "react";
-import {
-  catchError,
-  filter,
-  map,
-  startWith,
-  switchMap,
-  take
-} from "rxjs/operators";
-import { BehaviorSubject, combineLatest, merge, of } from "rxjs";
+import { catchError, map, startWith } from "rxjs/operators";
+import { combineLatest, of } from "rxjs";
 import { componentFromStream } from "@dcos/data-service";
 
 import ConfigurationMapSection from "#SRC/js/components/ConfigurationMapSection";
-import { Package } from "#SRC/js/data/cosmos/Package";
 import { UIMetadata } from "#SRC/js/data/ui-update/UIMetadata";
 
 import { rollbackUI, updateUI } from "#PLUGINS/ui-update/commands";
 import AvailableUpdateRow from "#PLUGINS/ui-update/components/AvailableUpdateRow";
 import InstalledVersion from "#PLUGINS/ui-update/components/InstalledVersion";
 import InitialScreen from "#PLUGINS/ui-update/components/InitialScreen";
 import FallbackScreen from "#PLUGINS/ui-update/components/FallbackScreen";
+import { UIAction } from "#PLUGINS/ui-update/types/UIAction";
 import {
-  queryCosmosForUIVersions,
-  queryUIServiceForMetadata
-} from "#PLUGINS/ui-update/queries";
-import {
-  EMPTY_ACTION,
-  UIAction,
-  UIActions,
-  UIActionType
-} from "#PLUGINS/ui-update/types/UIAction";
-import { versionUpdateAvailable } from "#PLUGINS/ui-update/utils";
-
-const uiServiceActions$ = new BehaviorSubject<UIAction>(EMPTY_ACTION);
+  getAction$,
+  getUiMetadata$,
+  getUpdateAvailable$
+} from "#PLUGINS/ui-update/streams";
+import { FormattedPackageVersion } from "#PLUGINS/ui-update/types/FormattedPackageVersion";
 
 function handleRollbackUI() {
-  uiServiceActions$.next({
-    type: UIActionType.Reset,
-    action: UIActions.Started,
-    value: ""
-  });
-  rollbackUI()
-    .pipe(take(1))
-    .subscribe(
-      result => {
-        uiServiceActions$.next({
-          type: UIActionType.Reset,
-          action: UIActions.Completed,
-          value: result.data.resetDCOSUI || ""
-        });
-      },
-      error => {
-        uiServiceActions$.next({
-          type: UIActionType.Reset,
-          action: UIActions.Error,
-          value: error.message
-        });
-      }
-    );
+  rollbackUI();
```

Comment:
Can't we directly pass rollbackUI whereever we are passing handleRollbackUI (same for handleUpdateClick)

Reply:
no, the click handler would send eventArgs as the first parameter to `rollbackUI`. Which would take them as the `delayMs`.

Once race-condition is fixed, yes we could easily refactor this to be passed directly.

Reply:
ah okay :)